src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java

```diff
@@ -189,33 +189,48 @@ public void setAllowEmptyLoopBody(boolean allowEmptyLoopBody) {
 
     @Override
     public void visitToken(DetailAST ast) {
-        final DetailAST slistAST = ast.findFirstToken(TokenTypes.SLIST);
-        boolean isElseIf = false;
-        if (ast.getType() == TokenTypes.LITERAL_ELSE
-            && ast.findFirstToken(TokenTypes.LITERAL_IF) != null) {
-            isElseIf = true;
+        final int type = ast.getType();
+        final boolean hasNoSlist = ast.findFirstToken(TokenTypes.SLIST) == null;
+        boolean hasViolation = hasNoSlist && !isSkipStatement(ast);
+        switch (type) {
```

Comment:
@strkkk please move this `switch` to a new method. After that the whole method can be just
```
        if (hasNoSlist && !isSkipStatement(ast) && needBraces(ast)) {
            log(ast.getLineNo(), MSG_KEY_NEED_BRACES, ast.getText());
        }
```
and the `needBraces` method will be more readable without strnge
`boolean = boolean && expr`

Reply:
In other Checks we try to keep log of violations in visitToken methods, it kind force author to think on design of Check in stateless approach. Only very complicated Checks do logging in some inner methods, but still we discourage deep nesting of such logs.

Reply:
@pbludov 
It can be done, but I don't know what name to choose for this method.
`needBraces` is not clear for me, actually it is just bunch of additional checks for specific token types.
Do you have any ideas?

Reply:
done